### PR TITLE
Basic Storyteller webapp

### PR DESCRIPTION
**Pull request by LiadAzulay**

### CODE REVIEW

File: projects/storyteller/webapp/proto/data.proto [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+  // Optional tagline at the bottom
+  string tagline = 8;
+  double hourly_rate_usd = 9;
+}
+
+message InvoiceReceiver {
+  string name = 1;
+  string street_and_number = 2;
+  string city_state_country = 3;
+}
```
Comment (webdevelopland) [inline projects/storyteller/webapp/proto/data.proto]: It's good practice to end each file with new empty line.
Temporally you can use github to fix it.
When you created new PR, you can check your code and then push new commit with fixes.

Github highlights `end without new line`:
https://i.imgur.com/dDk4beZ.jpg

File: projects/storyteller/webapp/src/app/app.component.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  selector: 'app-root',
+  templateUrl: './app.component.html',
+  styleUrls: ['./app.component.scss']
+})
+export class AppComponent {
+  storylist: StoryList[];
+
+
+  constructor(
+    //private encodingService: EncodingService,
```
Comment (webdevelopland) [inline projects/storyteller/webapp/src/app/app.component.ts]: Please try to not leave commented code in PR

File: projects/storyteller/webapp/src/app/import/firebase/firebase.module.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import { NgModule } from '@angular/core';
+import { AngularFireModule } from 'angularfire2';
+import { AngularFireAuth, AngularFireAuthModule } from 'angularfire2/auth';
+import { AngularFireDatabase } from 'angularfire2/database';
+import { AngularFirestoreModule, FirestoreSettingsToken } from 'angularfire2/firestore';
+
+import { config } from './config';
+import { FirestoreSettingsToken } from '@angular/fire/firestore';
```
Comment (webdevelopland) [inline projects/storyteller/webapp/src/app/import/firebase/firebase.module.ts]: `FirestoreSettingsToken` already imported on line 5.
Line 8 is excess.

File: projects/storyteller/webapp/proto/data.proto (hunk 1 above)
Comment (webdevelopland) [reply projects/storyteller/webapp/proto/data.proto]: Example of file with new empty line:
https://i.imgur.com/yRdmDZK.jpg

File: projects/storyteller/webapp/src/app/app.component.html [hunk 4]
```diff
@@ -0,0 +1,13 @@
+<div *ngIf="!storylist">
+    Loading
+</div>
+<div *ngIf="storylist">
+    <div *ngFor="let storylistChild of storylist">
```
Comment (webdevelopland) [inline projects/storyteller/webapp/src/app/app.component.html]: Usually each project has one intent size for all files.
Intent size means how much spaces is contained in one tab.

It's set here:
https://github.com/hasadna/hasadna/blob/finishing/projects/storyteller/webapp/.editorconfig#L7

The project has 2 spaces intent size. But the file has 4 per tab.
https://i.imgur.com/s6afFx2.jpg

File: projects/storyteller/webapp/tslint.json [hunk 5]
```diff
@@ -0,0 +1,131 @@
+{
+  "rulesDirectory": [
+    "codelyzer"
+  ],
+  "rules": {
```
Comment (webdevelopland) [inline projects/storyteller/webapp/tslint.json]: It's `tslint` rules, description on how should be written typescript code.
And good thing is it's not just declaration. `tslint` can fix errors automatically.

To see errors:
```
ng lint
```

To fix errors:
```
ng lint --fix
```
(or `ng lint -fix`, if you have earlier version of ng)

If you don't have ng:
```
npm install -g @angular/cli
```

Comment (webdevelopland) [reply projects/storyteller/webapp/tslint.json]: BTW, tslint can fix `end without new line` in `.ts` files.

Comment (webdevelopland) [reply projects/storyteller/webapp/tslint.json]: You can see here what each rule means:
https://palantir.github.io/tslint/rules/

Comment (LiadAzulay) [reply projects/storyteller/webapp/tslint.json]: thanks a lot Vadim!!!

File: projects/storyteller/webapp/src/app/app.component.html (hunk 4 above)
Comment (LiadAzulay) [reply projects/storyteller/webapp/src/app/app.component.html]: I will change my visual studio settings right now

File: projects/storyteller/webapp/proto/data.proto (hunk 1 above)
Comment (LiadAzulay) [reply projects/storyteller/webapp/proto/data.proto]: like python, I will pay attention to it from now on

**Pull request merged**